src/main/java/fi/aalto/cs/apluscourses/ui/repl/ReplConfigurationForm.java

```diff
@@ -0,0 +1,137 @@
+    this.model = model;
+    dontShowThisWindowCheckBox.setSelected(showREPLConfigWindow);
+
+    infoTextLabel.setText(INFOLABEL_TEXT);
+
+    addFileChooser("Choose Working Directory", workingDirectoryField, model.getProject());
+    workingDirectoryField.setText(model.getModuleWorkingDirectory());
+
+    model.getModuleNames().forEach(moduleName -> moduleComboBox.addItem(moduleName));
+    moduleComboBox.setSelectedItem(model.getTargetModuleName());
+    moduleComboBox.setEnabled(true);
+    moduleComboBox.setRenderer(new ModuleComboBoxListRenderer());
+  }
+
+  private void addFileChooser(
+      @NotNull final String title,
+      @NotNull final TextFieldWithBrowseButton textField,
+      @NotNull final Project project) {
+    final FileChooserDescriptor fileChooserDescriptor = new FileChooserDescriptor(
+        false, true, false,
+        false, false, false) {
+      @Override
+      public boolean isFileVisible(VirtualFile file, boolean showHiddenFiles) {
+        return super.isFileVisible(file, showHiddenFiles) && file.isDirectory();
+      }
+    };
+    fileChooserDescriptor.setTitle(title);
+    textField.addBrowseFolderListener(title, null, project, fileChooserDescriptor);
+  }
+
+  /**
+   * This method updates the embedded {@link ReplConfigurationFormModel} with the current state of
+   * the {@link ReplConfigurationForm} build.
+   */
+  public void updateModel() {
+    model.setTargetModuleName(requireNonNull(moduleComboBox.getSelectedItem()).toString());
+    model.setModuleWorkingDirectory(workingDirectoryField.getText());
+    showREPLConfigWindow = !dontShowThisWindowCheckBox.isSelected();
+  }
+
+  public void cancelReplStart() {
+    model.setStartRepl(false);
+  }
+
+  public JPanel getContentPane() {
+    return contentPane;
+  }
+
+  public ReplConfigurationFormModel getModel() {
+    return model;
+  }
+
+  public void setModel(ReplConfigurationFormModel model) {
+    this.model = model;
+  }
+
+  public TextFieldWithBrowseButton getWorkingDirectoryField() {
+    return workingDirectoryField;
+  }
+
+  public void setWorkingDirectoryField(
+      TextFieldWithBrowseButton workingDirectoryField) {
+    this.workingDirectoryField = workingDirectoryField;
+  }
+
+  public ComboBox getModuleComboBox() {
+    return moduleComboBox;
+  }
+
+  public void setModuleComboBox(ComboBox moduleComboBox) {
+    this.moduleComboBox = moduleComboBox;
+  }
+
+  public JCheckBox getDontShowThisWindowCheckBox() {
+    return dontShowThisWindowCheckBox;
+  }
+
+  public void setDontShowThisWindowCheckBox(JCheckBox dontShowThisWindowCheckBox) {
+    this.dontShowThisWindowCheckBox = dontShowThisWindowCheckBox;
+  }
+
+  public void setContentPane(JPanel contentPane) {
+    this.contentPane = contentPane;
+  }
+
+  public JLabel getInfoTextLabel() {
+    return infoTextLabel;
+  }
+
+  public void setInfoTextLabel(JLabel infoTextLabel) {
+    this.infoTextLabel = infoTextLabel;
+  }
+}
```

Comment:
Could remove all of these `setX` methods since they are unused (unless they are used by the GUI designer).